src/raft_server.c

```diff
@@ -609,7 +604,7 @@ int raft_recv_requestvote(raft_server_t* me_,
         node = raft_get_node(me_, vr->candidate_id);
 
     /* Reject request if we have a leader */
-    if (me->leader_id != -1 && me->leader_id != raft_node_get_id(node) &&
+    if (me->leader_id != -1 && me->leader_id != vr->candidate_id &&
```

Comment:
Only some questions / assertions to confirm understanding. No change requested.

Is the change here motivated by the empty configuration case where node will be NULL? And less motivated by the case in which the node is associated with the candidate_id in the RV message?
Either way it seems that this conditional will be false, given the test scenarios described in the commit message and the calling test function - so the _should_grant_vote() change is the one that seems to enable those scenarios to function properly.

Reply:
@kccain, yes for both questions. This change was added later when I proof-read the patch. I think we will remove the raft_node_t parameters of at least raft_recv_requestvote and raft_recv_appendentries in the future, probably when we remove the reuse of node IDs, for which we likely want to add node ID checks for incoming messages. The change here, however, seemed to fit the title of the patch, "vote despite membership", and not in conflict with the future plan. Hence I added it.